src/components/tooltip/daikin-tooltip.ts

```diff
@@ -0,0 +1,252 @@
+import {
+  autoUpdate,
+  computePosition,
+  flip,
+  offset,
+  shift,
+} from "@floating-ui/dom";
+import { cva } from "class-variance-authority";
+import { css, html, LitElement, unsafeCSS, type PropertyValues } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { createRef, ref, type Ref } from "lit/directives/ref.js";
+import tailwindStyles from "../../tailwind.css?inline";
+
+const cvaTooltip = cva(
+  [
+    "absolute",
+
+    "p-3",
+    "justify-center",
+    "items-center",
+    "rounded",
+    "w-max",
+    "top-0",
+    "left-0",
+    "max-w-[312px]",
+
+    "text-sm",
+    "font-daikinSerif",
+    "font-normal",
+    "not-italic",
+    "leading-5",
+  ],
+  {
+    variants: {
+      variant: {
+        light: [
+          "border",
+          "border-solid",
+          "border-daikinNeutral-800",
+          "bg-[#ffffffe6]",
+          "text-black",
+        ],
+        dark: ["bg-[#414141e6]", "text-white"],
+      },
+      open: {
+        true: ["visible", "opacity-100"],
+        false: ["invisible", "opacity-0"],
+      },
+    },
+  }
+);
+
+/**
+ * A tooltip component is used to show brief information when a user interacts with an element.
+ *
+ * @fires beforetoggle - _Cancellable._ A custom event emitted when the tooltip is about to be opened or closed by user interaction.
+ *
+ * @fires toggle - A custom event emitted when the tooltip is opened or closed.
+ *
+ * @slot - A slot for the element to which the tooltip is attached (the trigger element).
+ *
+ * @slot tooltip - A slot for the tooltip content.
+ *
+ * @example
+ *
+ * ```html
+ *  </daikin-tooltip>
+ *    <span slot="tooltip">This is a message</span>
+ *    <span>hover me</span>
+ *  </daikin-tooltip>
+ *
+ *  </daikin-tooltip description="This is a message">
+ *    <span>hover me</span>
+ *  </daikin-tooltip>
+ * ```
```

Comment:
```suggestion
 * @example
 *
 * ```html
 * <daikin-tooltip>
 *   <span slot="tooltip">This is a message</span>
 *   <span>Hover me</span>
 * </daikin-tooltip>
 * ```
 *
 * ```html
 * <daikin-tooltip description="This is a message">
 *   <span>Hover me</span>
 * </daikin-tooltip>
 * ```
```

Reply:
@rinjInTokyo 
Didn't see the change.